chainermn/communicators/_base.py

```diff
@@ -15,6 +19,19 @@ def rank(self):
     def size(self):
         return self.mpi_comm.size
 
+    def send(self, array, dest, tag):
+        shape = numpy.array(array.shape, dtype='int')
+        buf = _memory_utility.array_to_buffer_object(array.astype('float32'))
```

Comment:
As forcing types is a little dangerous, how about check that the input array is `float32` by `assert`?